workflow/scripts/feature_extraction/extract.py

```
@@ -0,0 +1,372 @@
+
+        try:
+            # Find all non-empty feature dataframes in the globbed list and concatenate them
+            dataset_features = pd.concat(non_empty_dfs(sample_feature_files))
+            # Sort the dataframes by the sample ID column
+            dataset_features = dataset_features.sort_values(by="SampleID")
+            # Save out the combined feature dataframe
+            dataset_features.to_csv(dataset_features_path, index=False)
+
+        except ValueError:
+            # Handle case where all dataframes are empty
+            logger.error(f"No non-empty dataframes found for {permutation} {region}.")
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)